compiler/src/cobalt/Parser/ExprParser.hs

```diff
@@ -152,31 +146,57 @@ forLoopGeneratorParser  = try $ L.indentBlock scn p
       rword "to"
       end <- aTerm
       symbol ")"
-      return (L.IndentMany Nothing (return . (For varName start end) . Block) statementParser)
+      return (L.IndentMany Nothing (return . (For varName start end) . BlockStmt) statementParser)
 
-identifierParser :: Parser Stmt
+identifierParser :: Parser Expr
 identifierParser = do
     name <- nameParser
     return $ Identifier name
 
 ifStatementParser :: Parser Stmt
 ifStatementParser  = do
-    ifSection   <- L.indentBlock scn ifP
-    elifSection <- optional $ L.indentBlock scn elifP
-    elseSection <- optional $ L.indentBlock scn elseP
-    return $ If ifSection elifSection elseSection
+    (condition, ifStatements) <- ifParser
+    elseSection <- elseParser
+    return $ If condition ifStatements elseSection
   where
-    ifP = do
-      rword "if"
-      condition  <- parens bExpr
-      return (L.IndentMany Nothing (return . (IfStatement condition) . Block) statementParser)
-    elifP = do
-      rword "elif"
-      condition  <- parens bExpr
-      return (L.IndentMany Nothing (return . (ElifStatement condition) . Block) statementParser)
-    elseP = do
-      rword "else"
-      return (L.IndentMany Nothing (return . (ElseStatement) . Block) statementParser)
+
+    controlParser = do
+        condition <- bExpr
+        rword "then"
+        return condition
+
+    ifParser = do
+        try $ L.indentBlock scn p
+          where
+            p = do
+                rword "if"
+                condition <- controlParser
+                return (L.IndentSome Nothing (return .  (condition, ) . BlockStmt) statementParser)
+
+    elseParser = do
+        elifElse
+      where
+        elifElse = do
+            elifBlock <- optional elifP
```

Comment:
This is just cosmetic, but it seems like `elifElse` does not need to be defined in where.

Reply:
Unless this can be done in a one liner isn't elifElse required? It recursively calls itself so it would need to be defined.

Reply:
I have just played with it for a while and I think it might be possible, but I am still not familiar with the indentation magic, so I cant verify it and there might be some big obstacle I don't see . But since it works well like it is I now think that we can leave it and eventually come back to it when doing some refactoring.

Reply:
I think you're right it can be done quite easily after looking again. 
Just remove the `elifElse` and make it recursively call `elseParser` instead.